### PR TITLE
Fix ordering of chunks where UI would not update

### PR DESCRIPTION
**Pull request by TimvdLippe**
Finally fixed the ordering again in the UI. We would only need to update according to the provided debug statement in `_diffMoved`.

Also I had to fix some issues with the service-worker, which was returning corrupt requests and some safe-guards with pullRequest being nulled which fired the ajax but then resulted in invalid responses.

Sorry for this fix-more-than-one-thing-PR, but these changes were required for me to test the ordering fixes :(

### CODE REVIEW

**Comment by txsmith** [comment]
The + button does not create a group anymore but instead creates an empty diff:
screenshot from 2016-05-22 15 01 04

**Comment by txsmith** [comment]
Confirmed working!

**Comment by eanker** [comment]
All pr's have 2 empty chunks before the pr changes.

File: src/pull-request/pull-request-chunk.html [hunk 1]
```diff
@@ -120,23 +123,50 @@ <h2 hidden$="[[_is(editable, title)]]">[[title]]</h2>
...
+          }
+        }
+      },
+      _diffMoved: function(e) {
+        if (e.detail) {
+          console.log('Moved from %d to %d', e.detail.newIndex, e.detail.oldIndex);
+        }
       },
-      _diffRemoved: function() {
+      _diffRemoved: function(e) {
```
Comment (eanker) [inline src/pull-request/pull-request-chunk.html:169]: Why this e

File: src/pull-request/pull-request-chunk.html [hunk 2]
```diff
@@ -120,23 +123,50 @@ <h2 hidden$="[[_is(editable, title)]]">[[title]]</h2>
...
+            this.fire('upgraded', {
+              file: changes.file,
+              diffId: changes.diffId
+            });
+          }
+        }
+      },
+      _diffMoved: function(e) {
+        if (e.detail) {
+          console.log('Moved from %d to %d', e.detail.newIndex, e.detail.oldIndex);
```
Comment (eanker) [inline src/pull-request/pull-request-chunk.html:166]: Why this console.log

**Comment by eanker** [comment]
lgtm

**Pull request merged**